bug: Error while installing xcuitest driver

### Do I have the most recent component updates?

- [X] I use the most recent available driver/plugin and server versions

### Is the component officially supported by the Appium team?

- [X] I have verified the component repository is present under the Appium organization in GitHub

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

## Description

When I execute the following command:

```shell
appium driver install xcuitest
```

I get the following Error on my latest Mac OS M1 machine:

```shell
✖ Checking if 'appium-xcuitest-driver' is compatible
Error: npm command 'info xcuitest peerDependencies dependencies --json' failed with code 1.

STDOUT:
{
  "error": {
    "code": "E404",
    "summary": "Not Found - GET https://registry.npmjs.org/xcuitest - Not found",
    "detail": "\n 'xcuitest@*' is not in this registry.\n\nNote that you can also install from a\ntarball, folder, http url, or git url."
  }
}

STDERR:
npm ERR! code E404
npm ERR! 404 Not Found - GET https://registry.npmjs.org/xcuitest - Not found
npm ERR! 404 
npm ERR! 404  'xcuitest@*' is not in this registry.
npm ERR! 404 
npm ERR! 404 Note that you can also install from a
npm ERR! 404 tarball, folder, http url, or git url.

npm ERR! A complete log of this run can be found in: /Users/wasiqbhamla/.npm/_logs/2024-01-04T07_27_29_678Z-debug-0.log
```

Following is the output in the log file mentioned in the above Error:

```shell
0 verbose cli /Users/wasiqbhamla/.nvm/versions/node/v18.19.0/bin/node /Users/wasiqbhamla/.nvm/versions/node/v18.19.0/bin/npm
1 info using npm@10.2.5
2 info using node@v18.19.0
3 timing npm:load:whichnode Completed in 0ms
4 timing config:load:defaults Completed in 1ms
5 timing config:load:file:/Users/wasiqbhamla/.nvm/versions/node/v18.19.0/lib/node_modules/npm/npmrc Completed in 0ms
6 timing config:load:builtin Completed in 0ms
7 timing config:load:cli Completed in 1ms
8 timing config:load:env Completed in 1ms
9 timing config:load:file:/Users/wasiqbhamla/Documents/github/.npmrc Completed in 0ms
10 timing config:load:project Completed in 1ms
11 timing config:load:file:/Users/wasiqbhamla/.npmrc Completed in 1ms
12 timing config:load:user Completed in 1ms
13 timing config:load:file:/Users/wasiqbhamla/.nvm/versions/node/v18.19.0/etc/npmrc Completed in 0ms
14 timing config:load:global Completed in 0ms
15 timing config:load:setEnvs Completed in 1ms
16 timing config:load Completed in 6ms
17 timing npm:load:configload Completed in 6ms
18 timing config:load:flatten Completed in 1ms
19 timing npm:load:mkdirpcache Completed in 0ms
20 timing npm:load:mkdirplogs Completed in 0ms
21 verbose title npm info xcuitest peerDependencies dependencies
22 verbose argv "info" "xcuitest" "peerDependencies" "dependencies" "--json"
23 timing npm:load:setTitle Completed in 4ms
24 timing npm:load:display Completed in 1ms
25 verbose logfile logs-max:10 dir:/Users/wasiqbhamla/.npm/_logs/2024-01-04T07_27_29_678Z-
26 verbose logfile /Users/wasiqbhamla/.npm/_logs/2024-01-04T07_27_29_678Z-debug-0.log
27 timing npm:load:logFile Completed in 4ms
28 timing npm:load:timers Completed in 0ms
29 timing npm:load:configScope Completed in 0ms
30 timing npm:load Completed in 22ms
31 silly logfile start cleaning logs, removing 1 files
32 silly logfile done cleaning log files
33 http fetch GET 404 https://registry.npmjs.org/xcuitest 1088ms (cache skip)
34 timing command:info Completed in 1091ms
35 verbose stack HttpErrorGeneral: 404 Not Found - GET https://registry.npmjs.org/xcuitest - Not found
35 verbose stack     at /Users/wasiqbhamla/.nvm/versions/node/v18.19.0/lib/node_modules/npm/node_modules/npm-registry-fetch/lib/check-response.js:95:15
35 verbose stack     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
35 verbose stack     at async RegistryFetcher.packument (/Users/wasiqbhamla/.nvm/versions/node/v18.19.0/lib/node_modules/npm/node_modules/pacote/lib/registry.js:91:19)
35 verbose stack     at async View.getData (/Users/wasiqbhamla/.nvm/versions/node/v18.19.0/lib/node_modules/npm/lib/commands/view.js:197:20)
35 verbose stack     at async View.exec (/Users/wasiqbhamla/.nvm/versions/node/v18.19.0/lib/node_modules/npm/lib/commands/view.js:105:28)
35 verbose stack     at async module.exports (/Users/wasiqbhamla/.nvm/versions/node/v18.19.0/lib/node_modules/npm/lib/cli-entry.js:61:5)
36 verbose statusCode 404
37 verbose pkgid xcuitest@*
38 verbose cwd /Users/wasiqbhamla/Documents/github
39 verbose Darwin 23.2.0
40 verbose node v18.19.0
41 verbose npm  v10.2.5
42 error code E404
43 error 404 Not Found - GET https://registry.npmjs.org/xcuitest - Not found
44 error 404
45 error 404  'xcuitest@*' is not in this registry.
46 error 404
46 error 404 Note that you can also install from a
47 error 404 tarball, folder, http url, or git url.
48 verbose exit 1
49 timing npm Completed in 1197ms
50 verbose code 1
51 error A complete log of this run can be found in: /Users/wasiqbhamla/.npm/_logs/2024-01-04T07_27_29_678Z-debug-0.log
```

### Expected Behavior

The `xcuitest` driver should get installed successfully like it happened with earlier version of Appium 2.1.x

### Minimal Reproducible Example

## On new Mac OS

Run `appium driver install xcuitest`

## On existing machine with xcuitest driver

First uninstall existing xcuitest driver and then run the above command again to install the driver.

### Environment

- Operating system: Mac OS Sonomo 14.2.1
- Appium server version (output of `appium --version`): 2.4.0
- Appium driver(s) and their version(s): xcuitest
- Appium plugin(s) and their version(s):
- Node.js version (output of `node --version`): 18.19.0
- `npm` version (output of `npm --version`): 10.2.5
- Last component(s) version which did _not_ exhibit the problem: Appium 2.3.0
- Platform and version under test: iOS 17.2
- Real device or emulator/simulator: NA


### Link to Appium Logs

_No response_

### Further Information

_No response_

--- a/packages/appium/lib/cli/extension-command.js
+++ b/packages/appium/lib/cli/extension-command.js
@@ -36,13 +36,17 @@ function receiptToManifest(receipt) {
 }
 
 /**
- * Fetches the remote extension version requirement
+ * Fetches the remote extension version requirements
  *
- * @param {string} installSpec Extension name (could also contain the version suffix)
+ * @param {string} pkgName Extension name
+ * @param {string} [pkgVer] Extension version (if not provided then the latest is assumed)
  * @returns {Promise<[string, string|null]>}
  */
-async function getRemoteExtensionVersionReq(installSpec) {
-  const allDeps = await npm.getPackageInfo(installSpec, ['peerDependencies', 'dependencies']);
+async function getRemoteExtensionVersionReq(pkgName, pkgVer) {
+  const allDeps = await npm.getPackageInfo(
+    `${pkgName}${pkgVer ? `@${pkgVer}` : ``}`,
+    ['peerDependencies', 'dependencies']
+  );
   const requiredVersionPair = _.flatMap(_.values(allDeps).map(_.toPairs))
     .find(([name]) => name === 'appium');
   return [npmPackage.version, requiredVersionPair ? requiredVersionPair[1] : null];
@@ -246,13 +250,13 @@ class ExtensionCliCommand {
    * @param {InstallViaNpmArgs} installViaNpmOpts
    * @returns {Promise<void>}
    */
-  async _checkInstallCompatibility({installSpec, pkgName, installType}) {
+  async _checkInstallCompatibility({installSpec, pkgName, pkgVer, installType}) {
     if (INSTALL_TYPE_NPM !== installType) {
       return;
     }
 
     await spinWith(this.isJsonOutput, `Checking if '${pkgName}' is compatible`, async () => {
-      const [serverVersion, extVersionRequirement] = await getRemoteExtensionVersionReq(installSpec);
+      const [serverVersion, extVersionRequirement] = await getRemoteExtensionVersionReq(pkgName, pkgVer);
       if (serverVersion && extVersionRequirement && !semver.satisfies(serverVersion, extVersionRequirement)) {
         throw this._createFatalError(
           `'${installSpec}' cannot be installed because the server version it requires (${extVersionRequirement}) ` +

